internal/venice-common/src/main/java/com/linkedin/venice/kafka/consumer/ApacheKafkaConsumer.java

```diff
@@ -152,6 +178,15 @@ public ConsumerRecords<byte[], byte[]> poll(long timeoutMs) {
     while (attemptCount <= consumerPollRetryTimes && !Thread.currentThread().isInterrupted()) {
       try {
         records = kafkaConsumer.poll(Duration.ofMillis(timeoutMs));
+        polledPubSubMessages.clear();
```

Comment:
Is it safe to reuse this collection? Since it will be returned by this function and other code will access it later, it seems a bit dangerous. Maybe it's fine in today's code, but not sure if it that will remain the case, so it could end up being a bit fragile?

Reply:
Yeah, it is safe in current setting, but a bit fragile in the future. I will create a local map for each poll.